tests/integration/p2p.go

```diff
@@ -451,7 +309,61 @@ func getAllP2PCollections(
 	assert.Equal(s.t, expectedCollections, cols)
 }
 
-// waitForSync waits for all given wait channels to receive an item signaling completion.
+// waitForUpdateEvents waits for all selected nodes to publish an update event.
+//
+// Expected document heads will be updated for all nodes that should receive network merges.
+func waitForUpdateEvents(
+	s *state,
+	nodeID immutable.Option[int],
+	collectionID int,
+) {
+	if len(s.nodeUpdateSubs) == 0 {
+		return // skip network testing
+	}
+
+	ctx, cancel := context.WithTimeout(s.ctx, subscriptionTimeout*10)
+	defer cancel()
+
+	for i := 0; i < len(s.nodes); i++ {
+		if nodeID.HasValue() && nodeID.Value() != i {
+			continue // node is not selected
+		}
```

Comment:
suggestion: The documentation says `waits for all selected nodes` but the function only receives one. You can probably do away with the loop and just get the node directly.

Reply:
If `nodeID` has no value it will wait for all nodes.